apps/www/_blog/2023-04-27-postgres-pluggable-strorage.mdx

```diff
@@ -0,0 +1,61 @@
+
+This implementation requires more work for a true pluggable storage solution. In particular:
+
+- The TAM API doesn’t provide much flexibility for working with indexes. For example, when you update a row, an index update is an all-or-nothing procedure. This was infamously highlighted by Uber in their “[switching from Postgres to MySQL](https://www.uber.com/blog/postgres-to-mysql-migration/)” in 2016.
+- Row versions need to be identified the pairing of a block number and offset number. This means you can’t implement useful functions, like an index-organized table or undo-based versioning.
+
+While It was a great start, it appears that the [work on ZHeap has stalled](https://twitter.com/andy_pavlo/status/1590703943176589312).
+
+Even without a full API for pluggable storage, there are a couple of solutions available on the market today:
+
+- **Citus’s** [Columnar Storage](https://docs.citusdata.com/en/v11.1/admin_guide/table_management.html#columnar-storage), which powers Microsoft’s [Hyperscale](https://azure.microsoft.com/en-us/updates/azure-database-for-postgresql-hyperscale-citus-is-now-available/) offering. Citus has a few [limitations](https://docs.citusdata.com/en/v11.1/admin_guide/table_management.html#limitations) due to the limitations of the current API.
+- EDB’s [Advanced Storage Pack](https://www.infoworld.com/article/3681112/3-key-features-in-edb-postgresql-15.html), which is appears to provide some enhancements to heap storage, although the source is not available for us to confirm.
+
+## The future of pluggable storage
+
+Right now, pluggable storage requires a a fork of Postgres. But there are several initiatives in the Postgres community that we’re excited about.
+
+- There have been community updates for [custom resource managers](https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=5c279a6d350205cc98f91fb8e1d3e4442a6b25d1) that expand WAL support for TAM.
+- The creators at [OrioleDB](https://github.com/orioledb/orioledb) started with a fork of Postgres and have been quietly upstreaming the code that is required to make pluggable storage a reality. They are solving many “[wicked Postgres problems](https://www.slideshare.net/AlexanderKorotkov/solving-postgresql-wicked-problems)” with [dual pointers](https://www.orioledata.com/blog/buffer-management/) instead of buffer mapping, row-level WAL instead of block-level WAL, undo logs instead of bloat-prone MVCC, and index-organized tables instead of heap. In Postgres 14 their patchset was about 5,000 lines of code (this represents the total number of changes required in their fork). In the upcoming Postgres 16 release, it is around 2,000 lines of code. That means that around 60% of the code is already committed to the Postgres core. (Disclosure: Supabase started [sponsoring OrioleDB](https://supabase.com/blog/supabase-series-b#where-were-going) in 2022).
```

Comment:
Lots of content here. Possibly break this out into a paragraph vs a bullet point